src/librustc_error_codes/error_codes/E0631.md

```diff
@@ -0,0 +1,29 @@
+This error indicates a type mismatch in closure arguments.
+
+Erroneous code example:
+
+```compile_fail,E0631
+fn test_strings(string_vec: Vec<String>) -> Vec<bool> {
```

Comment:
Instead of using `map` on a `String`, what about using a "simpler" code like this one:

```rust
fn foo<F: Fn(i32)>(f: F) {
}

fn main() {
    foo(|x: &str| {});
}
```

?